OpenEdX/DI/AppAssembly.swift

```diff
@@ -177,7 +177,8 @@ class AppAssembly: Assembly {
                 deepLinkManager: r.resolve(DeepLinkManager.self)!,
                 storage: r.resolve(CoreStorage.self)!,
                 api: r.resolve(API.self)!,
-                config: r.resolve(ConfigProtocol.self)!
+                config: r.resolve(ConfigProtocol.self)!,
+                segmentService: r.resolve(SegmentAnalyticsService.self)
```

Comment:
Could we use an Analytics abstraction and not rely on the Segment dependency here?
We are going to migrate all third-party services to plugins, and this dependency could make it harder.